pre_commit/meta_hooks/check_files_matches_any.py

```diff
@@ -0,0 +1,36 @@
+import re
+import sys
+
+import pre_commit.constants as C
+from pre_commit.clientlib import load_config
+from pre_commit.git import get_all_files
+
+
+def files_matches_any(filenames, include):
+    include_re = re.compile(include)
+    for filename in filenames:
+        if include_re.search(filename):
+            return True
+    return False
+
+
+def check_files_matches_any(config_file=None):
+    config = load_config(config_file or C.CONFIG_FILE)
```

Comment:
to be a valid hook, this needs to take `1..*` arguments and process them.  Let's add an argument parser similar to [this one](https://github.com/pre-commit/pre-commit-hooks/blob/e718847ccbc65cd72a89fefb4db61c55ec4d430d/pre_commit_hooks/check_vcs_permalinks.py#L27-L40)

Reply:
I've also been moving more to conventionally naming this function `main` -- the function itself makes sense given the context that the module name gives us